src/middlewared/middlewared/plugins/smb.py

```diff
@@ -549,6 +561,12 @@ def getparm(self, parm, section):
         if new['netbiosname'] and new['netbiosname'].lower() == new['workgroup'].lower():
             verrors.add('smb_update.netbiosname', 'NetBIOS and Workgroup must be unique')
 
+        if data['bindip']:
+            bindip_choices = [x[0] for x in (await self.bindip_choices()).items()]
```

Comment:
bindip_choices = list((await self.bindip_choices()).keys())

is simpler ; )

Reply:
Yeah, I'm having a Monday. :)